### PR TITLE
adding Indexing classes and Suggest test

### PR DESCRIPTION
**Pull request by felfilali**
I deleted the previous Draft after applying the recommended changes and made this new one in order to have a cleaner commit history

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -160,4 +160,5 @@ test {
     if (!project.hasProperty('includePerfTests')) {
         exclude '**/YelpReviewsTest.class'
     }
+    systemProperties System.properties
```
Comment (umeshdangat) [inline build.gradle]: not sure what this line does

Comment (felfilali) [reply build.gradle]: This transfers the -D params given to ./gradlew to sys.properties in this particular task (test). 
If removed, and you ran `./gradlew test -Dflag:my_flag  --tests "org.apache.platypus.server.YelpSuggestTest", System.properties("my_flag") will result in a NullPointerExcption

Comment (umeshdangat) [reply build.gradle]: ok, thanks for explaining

**Comment by felfilali** [comment]
@umeshdangat @sarthakn7 was able to run YelpReviewsTest on this branch with the big file. I encountered some issues due to disk space issues. I also made excluded YelpSuggestTest

**Pull request merged**